app/main.py

```diff
@@ -1 +1,21 @@
-# write your code here
+from app.cafe import Cafe
+from app.errors import (NotWearingMaskError,
+                        NotVaccinatedError,
+                        OutdatedVaccineError)
+
+
+def go_to_cafe(friends, cafe) -> str:
+    total = 0
+
+    for i in range(len(friends)):
+        try:
+            cafe.visit_cafe(friends[i])
+        except (NotVaccinatedError, OutdatedVaccineError):
+            return 'All friends should be vaccinated'
+        except NotWearingMaskError:
+            total += 1
+
+    if total:
```

Comment:
you can improve it by using ternary operators in cases. when you will have less strings to return, or returning variables, instead of strings
``` python
return f"Friends should buy {total} masks" if total else f"Friends can go to {cafe.name}"

```